taichi/transforms/simplify.cpp

```diff
@@ -275,8 +275,24 @@ class BasicBlockSimplify : public IRVisitor {
       stmt->insert_before_me(std::move(mul));
       stride_product *= stmt->strides[i];
     }
-    stmt->replace_with(sum.get());
-    stmt->insert_before_me(std::move(sum));
+    // Compare the result with 0 to make sure no overflow occurs under Debug Mode.
+    bool debug = stmt->get_kernel()->program.config.debug;
+    if (debug) {
+      auto zero = Stmt::make<ConstStmt>(LaneAttribute<TypedConstant>(0));
+      auto check_sum = Stmt::make<BinaryOpStmt>(BinaryOpType::cmp_ge, sum.get(), zero.get());
+      auto assert = Stmt::make<AssertStmt>(check_sum.get(), "The indices provided are too big!", std::vector<Stmt *>());
+      auto select = Stmt::make<TernaryOpStmt>(TernaryOpType::select, check_sum.get(), sum.get(), zero.get());
```

Comment:
Let's add this comment to explain why we need `select`:

Because Taichi's assertion is checked only after the execution of the whole kernel, when the linear index overflows and goes negative, we have to replace that with 0 to make sure that the rest of the kernel can still complete. Otherwise, Taichi would crash due to the illegal memory address.